packages/poseidon-proof/src/generate.ts

```diff
@@ -26,19 +25,18 @@ export default async function generate(
     scope: BytesLike | Hexable | number | bigint,
     snarkArtifacts?: SnarkArtifacts
 ): Promise<PoseidonProof> {
-    // If the Snark artifacts are not defined they will be automatically downloaded.
-    /* istanbul ignore next */
-    if (!snarkArtifacts) {
-        snarkArtifacts = await getSnarkArtifacts(preimages.length)
-    }
+    // allow user to override our artifacts
+    // if not explicitly provided: download them from unpkg if not already in local tmp folder
```

Comment:
Maybe worth not mentioning unpkg here as it could change. External code shouldn't know what kind of CDN is used under the hood. Also, it'd be hard to update all the external occurences.